plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -2754,44 +2754,47 @@ function _wc_delete_transients( $transients ) {
 				);
 			}
 
-			// Use a single query for better performance.
-			$wpdb->query(
-				$wpdb->prepare(
-					'DELETE FROM ' . $wpdb->options . ' WHERE option_name IN ( ' . implode( ', ', array_fill( 0, count( $options_to_clear ), '%s' ) ) . ' )',
-					$options_to_clear
-				)
-			);
-
-			// Lets clear our options data from the cache.
-			// We can batch delete if available, introduced in WP 6.0.0.
-			if ( ! wp_installing() && ! empty( $options_to_clear ) ) {
-				if ( function_exists( 'wp_cache_delete_multiple' ) ) {
-					wp_cache_delete_multiple( $options_to_clear, 'options' );
-				} else {
-					foreach ( $options_to_clear as $option_name ) {
-						wp_cache_delete( $option_name, 'options' );
-					}
-				}
-
-				// Also update alloptions cache if needed.
-				// This is required to prevent phantom transients from being returned.
-				$alloptions         = wp_load_alloptions( true );
-				$updated_alloptions = false;
+			if ( ! empty( $options_to_clear ) ) {
```

Comment:
I just wrapped the next actions with this condition. If there aren't any `$options_to_clear` we shouldn't clear anything.

Reply:
Optional: Just a nitpick, and not a blocker but it might be better/more readable to inverse this condition:

```suggestion
			if ( empty( $options_to_clear ) ) {
				return true;
			}
```